cli/src/main/java/com/devonfw/tools/ide/tool/graalvm/GraalVm.java

```diff
@@ -0,0 +1,54 @@
+package com.devonfw.tools.ide.tool.graalvm;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.environment.EnvironmentVariables;
+import com.devonfw.tools.ide.environment.EnvironmentVariablesType;
+import com.devonfw.tools.ide.tool.LocalToolCommandlet;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * {@link LocalToolCommandlet} for <a href="https://www.graalvm.org/">GraalVM</a>, an advanced JDK with ahead-of-time Native Image compilation.
+ */
+public class GraalVm extends LocalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public GraalVm(IdeContext context) {
+
+    super(context, "graalvm", Set.of(Tag.JAVA));
+  }
+
+  @Override
+  public Path getToolPath() {
+
+    return this.context.getSoftwareExtraPath().resolve(getName());
+  }
+
+  @Override
+  public VersionIdentifier getInstalledVersion() {
+
+    return super.getInstalledVersion(getToolPath());
+  }
+
+  @Override
+  public void run() {
+
+    super.install(true);
+  }

```

Comment:
I guess we can stay with the default now. However, we need to verify that the proper toolname is set (you might need to override `getBinaryName()`) as there is no `graalvm` binary but instead `java` (or `javac` or `gu`, etc.).
Also you most probably need to change the `PATH` to prepend with GRAALVM_HOME/bin or you need to make the `Path` see variable `binaryPath` in "super.run()" absolute instead of relative...
```suggestion
```